src/components/Cast/CastBlock.tsx

```diff
@@ -0,0 +1,122 @@
+import { PostStatus } from 'models/PostStatus'
+import { PostStructOutput } from '@big-whale-labs/seal-cred-posts-contract/dist/typechain/contracts/SCPostStorage'
+import { Wallet } from 'ethers'
+import { handleError, parseErrorText } from '@big-whale-labs/frontend-utils'
+import { space } from 'classnames/tailwind'
+import { useSnapshot } from 'valtio'
+import { useState } from 'preact/hooks'
+import BurnerWalletStore from 'stores/BurnerWalletStore'
+import PostIdsStatuses from 'stores/PostIdsStatuses'
+import PostStore from 'stores/PostStore'
+import TextArea from 'components/ui/TextArea'
+import TextareaInfo from 'components/Cast/TextareaInfo'
+import VerifyWallet from 'components/Cast/VerifyWallet'
+import getErrorMessage from 'helpers/getErrorMessage'
+import hasFarcasterBadge from 'helpers/hasFarcasterBadge'
+import walletStore from 'stores/WalletStore'
+
+export default function () {
+  const { status } = useSnapshot(BurnerWalletStore)
+  const [isLoading, setIsLoading] = useState(false)
+  const [error, setError] = useState<unknown>()
+  const [text, setText] = useState('')
+  const [waitBurner, setWaitBurner] = useState(false)
+
+  const maxLength = 279
+  const errorMessage = error ? parseErrorText(error) : ''
+
+  async function createPost() {
+    let currentAccount
+    const { privateKey } = BurnerWalletStore
+    const { account } = walletStore
+
+    if (privateKey) currentAccount = new Wallet(privateKey).address
+    if (account && hasFarcasterBadge(account)) currentAccount = account
+
+    setError(null)
+    try {
+      if (!currentAccount) return setWaitBurner(true)
+
+      setIsLoading(true)
+      if (PostIdsStatuses.lastUserPost)
+        delete PostIdsStatuses.lastUserPost[currentAccount]
+
+      BurnerWalletStore.status = 'Posting cast'
+      const result = await PostStore.createPost(text)
+      const posts = await PostStore.posts
+      const numberOfPosts = await PostStore.postsAmount
+
+      PostStore.postsAmount = Promise.resolve(numberOfPosts + result.length)
+      for (const { id, post, derivativeAddress, sender, timestamp } of result) {
+        PostStore.posts = Promise.resolve([
+          {
+            id,
+            post,
+            derivativeAddress,
+            sender,
+            timestamp,
+          } as PostStructOutput,
+          ...posts,
+        ])
+
+        const blockchainId = id.toNumber()
+        const status = PostStatus.pending
+
+        PostIdsStatuses.lastUserPost = {
+          [currentAccount]: {
+            blockchainId,
+            status,
+          },
+          ...PostIdsStatuses.lastUserPost,
```

Comment:
Hmm, what if the user already had `lastUserPost` and wrote another one from the same account, will the user have 2 same keys?